File: development/NET/samples/client/DataControl/VB/OpcForm_VS2003.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Threading
Imports Softing.OPCToolbox
Namespace DataControl
  ''' <summary>
  ''' Summary description for OpcForm.
  ''' </summary>
  Public Class OpcForm
    Inherits System.Windows.Forms.Form
    Private groupBox1 As System.Windows.Forms.GroupBox
    Private button1 As System.Windows.Forms.Button
    Private label1 As System.Windows.Forms.Label
    Private label2 As System.Windows.Forms.Label
    Private label3 As System.Windows.Forms.Label
    Private label4 As System.Windows.Forms.Label
    Private label5 As System.Windows.Forms.Label
    Private textBox1 As System.Windows.Forms.TextBox
    Private textBox2 As System.Windows.Forms.TextBox
    Private textBox3 As System.Windows.Forms.TextBox
    Private textBox4 As System.Windows.Forms.TextBox
    Private textBox5 As System.Windows.Forms.TextBox
    Private groupBox2 As System.Windows.Forms.GroupBox
    Private groupBox3 As System.Windows.Forms.GroupBox
    Private dataControl1 As Softing.OPCToolbox.Client.Control.DataControl
    ''' <summary>
    ''' Required designer variable.
    ''' </summary>
    Private components As System.ComponentModel.Container = Nothing
    Private m_outProc As OutProc = Nothing
    Private m_opcClient As OpcClient = Nothing
    Public Sub New(ByVal anOutProc As OutProc)
      InitializeComponent()
      dataControl1.Start()

      Try
        m_outProc = anOutProc

        m_opcClient = m_outProc.OpcClient
      Catch exc As Exception
        MessageBox.Show(exc.ToString())
      End Try
    End Sub

    ''' <summary>
    ''' Clean up any resources being used.
    ''' </summary>
    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
      If disposing Then
        If Not components Is Nothing Then
          components.Dispose()
        End If
      End If
      MyBase.Dispose(disposing)
    End Sub

#Region "Windows Form Designer generated code"
    ''' <summary>
    ''' Required method for Designer support - do not modify
    ''' the contents of this method with the code editor.
    ''' </summary>
    Private Sub InitializeComponent()
      Dim resources As New System.Resources.ResourceManager(GetType(OpcForm))
      Me.groupBox1 = New System.Windows.Forms.GroupBox()
      Me.groupBox3 = New System.Windows.Forms.GroupBox()
      Me.textBox4 = New System.Windows.Forms.TextBox()
      Me.label4 = New System.Windows.Forms.Label()
      Me.label5 = New System.Windows.Forms.Label()
      Me.textBox5 = New System.Windows.Forms.TextBox()
      Me.groupBox2 = New System.Windows.Forms.GroupBox()
      Me.textBox1 = New System.Windows.Forms.TextBox()
      Me.label1 = New System.Windows.Forms.Label()
      Me.label2 = New System.Windows.Forms.Label()
      Me.textBox2 = New System.Windows.Forms.TextBox()
      Me.label3 = New System.Windows.Forms.Label()
      Me.textBox3 = New System.Windows.Forms.TextBox()
      Me.button1 = New System.Windows.Forms.Button()
      Me.dataControl1 = New Softing.OPCToolbox.Client.Control.DataControl()
      Me.groupBox1.SuspendLayout()
      Me.groupBox3.SuspendLayout()
      Me.groupBox2.SuspendLayout()
      Me.SuspendLayout()
      ' 
      ' groupBox1
      ' 
      Me.groupBox1.Controls.Add(Me.groupBox3)
      Me.groupBox1.Controls.Add(Me.groupBox2)
      Me.groupBox1.Controls.Add(Me.button1)
			Me.groupBox1.Font = New System.Drawing.Font("Times New Roman", 9.75F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, Convert.ToByte(0))
      Me.groupBox1.Location = New System.Drawing.Point(8, 8)
      Me.groupBox1.Name = "groupBox1"
      Me.groupBox1.Size = New System.Drawing.Size(312, 416)
      Me.groupBox1.TabIndex = 0
      Me.groupBox1.TabStop = False
      Me.groupBox1.Text = "Softing OPC Toolkit Demo OPC DA Server"
      ' 
      ' groupBox3
      ' 
      Me.groupBox3.Controls.Add(Me.textBox4)
      Me.groupBox3.Controls.Add(Me.label4)
      Me.groupBox3.Controls.Add(Me.label5)
      Me.groupBox3.Controls.Add(Me.textBox5)
			Me.groupBox3.Font = New System.Drawing.Font("Times New Roman", 9.75F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, Convert.ToByte(0))
      Me.groupBox3.Location = New System.Drawing.Point(8, 208)
      Me.groupBox3.Name = "groupBox3"
      Me.groupBox3.Size = New System.Drawing.Size(296, 152)
      Me.groupBox3.TabIndex = 3
      Me.groupBox3.TabStop = False
      Me.groupBox3.Text = "Subscription2"
      ' 
      ' textBox4
      ' 
      Me.textBox4.Location = New System.Drawing.Point(16, 48)
      Me.textBox4.Name = "textBox4"
      Me.textBox4.Size = New System.Drawing.Size(256, 22)
      Me.textBox4.TabIndex = 10
      Me.textBox4.Text = ""
      ' 
      ' label4
      ' 
      Me.label4.Location = New System.Drawing.Point(8, 24)
      Me.label4.Name = "label4"
      Me.label4.TabIndex = 5
      Me.label4.Text = "maths.sin"
      ' 
      ' label5
      ' 
      Me.label5.Location = New System.Drawing.Point(8, 80)
      Me.label5.Name = "label5"
      Me.label5.Size = New System.Drawing.Size(104, 24)
      Me.label5.TabIndex = 6
      Me.label5.Text = "time.local.second"
      ' 
      ' textBox5
      ' 
      Me.textBox5.Location = New System.Drawing.Point(16, 104)
      Me.textBox5.Name = "textBox5"
      Me.textBox5.Size = New System.Drawing.Size(256, 22)
      Me.textBox5.TabIndex = 11
      Me.textBox5.Text = ""
      ' 
      ' groupBox2
      ' 
      Me.groupBox2.Controls.Add(Me.textBox1)
      Me.groupBox2.Controls.Add(Me.label1)
      Me.groupBox2.Controls.Add(Me.label2)
      Me.groupBox2.Controls.Add(Me.textBox2)
      Me.groupBox2.Controls.Add(Me.label3)
      Me.groupBox2.Controls.Add(Me.textBox3)
			Me.groupBox2.Font = New System.Drawing.Font("Times New Roman", 9.75F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, Convert.ToByte(0))
      Me.groupBox2.Location = New System.Drawing.Point(8, 24)
      Me.groupBox2.Name = "groupBox2"
      Me.groupBox2.Size = New System.Drawing.Size(296, 176)
      Me.groupBox2.TabIndex = 2
      Me.groupBox2.TabStop = False
      Me.groupBox2.Text = "Subscription1"
      ' 
      ' textBox1
      ' 
      Me.textBox1.Location = New System.Drawing.Point(16, 40)
      Me.textBox1.Name = "textBox1"
      Me.textBox1.Size = New System.Drawing.Size(256, 22)
      Me.textBox1.TabIndex = 7
      Me.textBox1.Text = ""
      ' 
      ' label1
      ' 
      Me.label1.Location = New System.Drawing.Point(8, 24)
      Me.label1.Name = "label1"
      Me.label1.Size = New System.Drawing.Size(120, 16)
      Me.label1.TabIndex = 2
      Me.label1.Text = "increment.UI1"
      ' 
      ' label2
      ' 
      Me.label2.Location = New System.Drawing.Point(8, 64)
      Me.label2.Name = "label2"
      Me.label2.Size = New System.Drawing.Size(128, 23)
      Me.label2.TabIndex = 3
      Me.label2.Text = "increment.DATE"
      ' 
      ' textBox2
      ' 
      Me.textBox2.Location = New System.Drawing.Point(16, 88)
      Me.textBox2.Name = "textBox2"
      Me.textBox2.Size = New System.Drawing.Size(256, 22)
      Me.textBox2.TabIndex = 8
      Me.textBox2.Text = ""
      ' 
      ' label3
      ' 
      Me.label3.Location = New System.Drawing.Point(8, 112)
      Me.label3.Name = "label3"
      Me.label3.Size = New System.Drawing.Size(128, 23)
      Me.label3.TabIndex = 4
      Me.label3.Text = "increment.BSTR array"
      ' 
      ' textBox3
      ' 
      Me.textBox3.Location = New System.Drawing.Point(16, 136)
      Me.textBox3.Name = "textBox3"
      Me.textBox3.Size = New System.Drawing.Size(256, 22)
      Me.textBox3.TabIndex = 9
      Me.textBox3.Text = ""
      ' 
      ' button1
      ' 
			Me.button1.Font = New System.Drawing.Font("Times New Roman", 9.75F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, Convert.ToByte(0))
      Me.button1.Location = New System.Drawing.Point(32, 376)
      Me.button1.Name = "button1"
      Me.button1.TabIndex = 0
      Me.button1.Text = "Bind"
      AddHandler Me.button1.Click, AddressOf Me.button1_Click
      ' 
      ' dataControl1
      ' 
      Me.dataControl1.Capacity = 16
      Me.dataControl1.BinaryLicenseDa = ""
      Me.dataControl1.BinaryLicenseXmlDa = ""
      Me.dataControl1.Session = (DirectCast((resources.GetObject("dataControl1.Session")), Softing.OPCToolbox.Client.Control.ControlDaSession))

      ' 
      ' OpcForm
      ' 
      Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
      Me.ClientSize = New System.Drawing.Size(328, 430)
      Me.Controls.Add(Me.groupBox1)
      Me.Icon = (DirectCast((resources.GetObject("$this.Icon")), System.Drawing.Icon))
      Me.Name = "OpcForm"
      Me.Text = "DataControl Demo"
      AddHandler Me.Closing, AddressOf Me.OpcForm_Closing
      Me.groupBox1.ResumeLayout(False)
      Me.groupBox3.ResumeLayout(False)
      Me.groupBox2.ResumeLayout(False)
      Me.ResumeLayout(False)
    End Sub
#End Region

    ''' <summary>
    ''' The main entry point for the application.
    ''' </summary>
    Shared Sub Main()
      Dim outProc As New OutProc()

      Try
        outProc.CreateOpcClient()
        Dim opcClient As OpcClient = outProc.OpcClient

        Dim opcForm As New OpcForm(outProc)

        Application.Run(opcForm)

        opcClient = Nothing
      Catch exc As Exception
        MessageBox.Show(exc.ToString())
      End Try

    End Sub

    Private Sub button1_Click(ByVal sender As Object, ByVal e As System.EventArgs)

      button1.Enabled = False

      Try
        textBox1.DataBindings.Add(New Binding("Text", dataControl1.GetItem("Subscription1", "increment.UI1"), "ValueToString"))
        textBox2.DataBindings.Add(New Binding("Text", dataControl1.GetItem("Subscription1", "increment.DATE"), "ValueToString"))
        textBox3.DataBindings.Add(New Binding("Text", dataControl1.GetItem("Subscription1", "increment.BSTR array"), "ValueToString"))
        textBox4.DataBindings.Add(New Binding("Text", dataControl1.GetItem("Subscription2", "maths.sin"), "ValueToString"))
        textBox5.DataBindings.Add(New Binding("Text", dataControl1.GetItem("Subscription2", "time.local.second"), "ValueToString"))
      Catch exc As Exception
        MessageBox.Show(exc.ToString())
      End Try
    End Sub

    Private Sub OpcForm_Closing(ByVal sender As Object, ByVal e As System.ComponentModel.CancelEventArgs)
      If Not dataControl1 Is Nothing Then
        dataControl1.[Stop]()
      End If
    End Sub

  End Class
End Namespace

File: development/NET/samples/client/DataControl/CS/OpcForm_VS2003.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Threading;
using Softing.OPCToolbox;

namespace DataControl
{
	/// <summary>
	/// Summary description for OpcForm.
	/// </summary>
	public class OpcForm : System.Windows.Forms.Form
	{
		private System.Windows.Forms.GroupBox groupBox1;
		private System.Windows.Forms.Button button1;
		private System.Windows.Forms.Label label1;
		private System.Windows.Forms.Label label2;
		private System.Windows.Forms.Label label3;
		private System.Windows.Forms.Label label4;
		private System.Windows.Forms.Label label5;
		private System.Windows.Forms.TextBox textBox1;
		private System.Windows.Forms.TextBox textBox2;
		private System.Windows.Forms.TextBox textBox3;
		private System.Windows.Forms.TextBox textBox4;
		private System.Windows.Forms.TextBox textBox5;
		private System.Windows.Forms.GroupBox groupBox2;
		private System.Windows.Forms.GroupBox groupBox3;
		private Softing.OPCToolbox.Client.Control.DataControl dataControl1;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;
		private OutProc m_outProc = null;
		private OpcClient m_opcClient = null;
		public OpcForm(OutProc anOutProc)
		{
			InitializeComponent();
			dataControl1.Start();

			try
			{
				m_outProc = anOutProc;

				m_opcClient = m_outProc.OpcClient;
			}
			catch(Exception exc)
			{				
				MessageBox.Show(exc.ToString());
			}	//	end try...catch
		}	//	end constructor

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if (components != null) 
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			System.Resources.ResourceManager resources = new System.Resources.ResourceManager(typeof(OpcForm));
			this.groupBox1 = new System.Windows.Forms.GroupBox();
			this.groupBox3 = new System.Windows.Forms.GroupBox();
			this.textBox4 = new System.Windows.Forms.TextBox();
			this.label4 = new System.Windows.Forms.Label();
			this.label5 = new System.Windows.Forms.Label();
			this.textBox5 = new System.Windows.Forms.TextBox();
			this.groupBox2 = new System.Windows.Forms.GroupBox();
			this.textBox1 = new System.Windows.Forms.TextBox();
			this.label1 = new System.Windows.Forms.Label();
			this.label2 = new System.Windows.Forms.Label();
			this.textBox2 = new System.Windows.Forms.TextBox();
			this.label3 = new System.Windows.Forms.Label();
			this.textBox3 = new System.Windows.Forms.TextBox();
			this.button1 = new System.Windows.Forms.Button();
			this.dataControl1 = new Softing.OPCToolbox.Client.Control.DataControl();
			this.groupBox1.SuspendLayout();
			this.groupBox3.SuspendLayout();
			this.groupBox2.SuspendLayout();
			this.SuspendLayout();
			// 
			// groupBox1
			// 
			this.groupBox1.Controls.Add(this.groupBox3);
			this.groupBox1.Controls.Add(this.groupBox2);
			this.groupBox1.Controls.Add(this.button1);
			this.groupBox1.Font = new System.Drawing.Font("Times New Roman", 9.75F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.groupBox1.Location = new System.Drawing.Point(8, 8);
			this.groupBox1.Name = "groupBox1";
			this.groupBox1.Size = new System.Drawing.Size(312, 416);
			this.groupBox1.TabIndex = 0;
			this.groupBox1.TabStop = false;
			this.groupBox1.Text = "Softing OPC Toolkit Demo OPC DA Server";
			// 
			// groupBox3
			// 
			this.groupBox3.Controls.Add(this.textBox4);
			this.groupBox3.Controls.Add(this.label4);
			this.groupBox3.Controls.Add(this.label5);
			this.groupBox3.Controls.Add(this.textBox5);
			this.groupBox3.Font = new System.Drawing.Font("Times New Roman", 9.75F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.groupBox3.Location = new System.Drawing.Point(8, 208);
			this.groupBox3.Name = "groupBox3";
			this.groupBox3.Size = new System.Drawing.Size(296, 152);
			this.groupBox3.TabIndex = 3;
			this.groupBox3.TabStop = false;
			this.groupBox3.Text = "Subscription2";
			// 
			// textBox4
			// 
			this.textBox4.Location = new System.Drawing.Point(16, 48);
			this.textBox4.Name = "textBox4";
			this.textBox4.Size = new System.Drawing.Size(256, 22);
			this.textBox4.TabIndex = 10;
			this.textBox4.Text = "";
			// 
			// label4
			// 
			this.label4.Location = new System.Drawing.Point(8, 24);
			this.label4.Name = "label4";
			this.label4.TabIndex = 5;
			this.label4.Text = "maths.sin";
			// 
			// label5
			// 
			this.label5.Location = new System.Drawing.Point(8, 80);
			this.label5.Name = "label5";
			this.label5.Size = new System.Drawing.Size(104, 24);
			this.label5.TabIndex = 6;
			this.label5.Text = "time.local.second";
			// 
			// textBox5
			// 
			this.textBox5.Location = new System.Drawing.Point(16, 104);
			this.textBox5.Name = "textBox5";
			this.textBox5.Size = new System.Drawing.Size(256, 22);
			this.textBox5.TabIndex = 11;
			this.textBox5.Text = "";
			// 
			// groupBox2
			// 
			this.groupBox2.Controls.Add(this.textBox1);
			this.groupBox2.Controls.Add(this.label1);
			this.groupBox2.Controls.Add(this.label2);
			this.groupBox2.Controls.Add(this.textBox2);
			this.groupBox2.Controls.Add(this.label3);
			this.groupBox2.Controls.Add(this.textBox3);
			this.groupBox2.Font = new System.Drawing.Font("Times New Roman", 9.75F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.groupBox2.Location = new System.Drawing.Point(8, 24);
			this.groupBox2.Name = "groupBox2";
			this.groupBox2.Size = new System.Drawing.Size(296, 176);
			this.groupBox2.TabIndex = 2;
			this.groupBox2.TabStop = false;
			this.groupBox2.Text = "Subscription1";
			// 
			// textBox1
			// 
			this.textBox1.Location = new System.Drawing.Point(16, 40);
			this.textBox1.Name = "textBox1";
			this.textBox1.Size = new System.Drawing.Size(256, 22);
			this.textBox1.TabIndex = 7;
			this.textBox1.Text = "";
			// 
			// label1
			// 
			this.label1.Location = new System.Drawing.Point(8, 24);
			this.label1.Name = "label1";
			this.label1.Size = new System.Drawing.Size(120, 16);
			this.label1.TabIndex = 2;
			this.label1.Text = "increment.UI1";
			// 
			// label2
			// 
			this.label2.Location = new System.Drawing.Point(8, 64);
			this.label2.Name = "label2";
			this.label2.Size = new System.Drawing.Size(128, 23);
			this.label2.TabIndex = 3;
			this.label2.Text = "increment.DATE";
			// 
			// textBox2
			// 
			this.textBox2.Location = new System.Drawing.Point(16, 88);
			this.textBox2.Name = "textBox2";
			this.textBox2.Size = new System.Drawing.Size(256, 22);
			this.textBox2.TabIndex = 8;
			this.textBox2.Text = "";
			// 
			// label3
			// 
			this.label3.Location = new System.Drawing.Point(8, 112);
			this.label3.Name = "label3";
			this.label3.Size = new System.Drawing.Size(128, 23);
			this.label3.TabIndex = 4;
			this.label3.Text = "increment.BSTR array";
			// 
			// textBox3
			// 
			this.textBox3.Location = new System.Drawing.Point(16, 136);
			this.textBox3.Name = "textBox3";
			this.textBox3.Size = new System.Drawing.Size(256, 22);
			this.textBox3.TabIndex = 9;
			this.textBox3.Text = "";
			// 
			// button1
			// 
			this.button1.Font = new System.Drawing.Font("Times New Roman", 9.75F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.button1.Location = new System.Drawing.Point(32, 376);
			this.button1.Name = "button1";
			this.button1.TabIndex = 0;
			this.button1.Text = "Bind";
			this.button1.Click += new System.EventHandler(this.button1_Click);
			// 
			// dataControl1
			// 
			this.dataControl1.Capacity = 16;
			this.dataControl1.BinaryLicenseDa = "";
			this.dataControl1.BinaryLicenseXmlDa = "";
			this.dataControl1.Session = ((Softing.OPCToolbox.Client.Control.ControlDaSession)(resources.GetObject("dataControl1.Session")));

			// 
			// OpcForm
			// 
			this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
			this.ClientSize = new System.Drawing.Size(328, 430);
			this.Controls.Add(this.groupBox1);
			this.Icon = ((System.Drawing.Icon)(resources.GetObject("$this.Icon")));
			this.Name = "OpcForm";
			this.Text = "DataControl Demo";
			this.Closing += new System.ComponentModel.CancelEventHandler(this.OpcForm_Closing);
			this.groupBox1.ResumeLayout(false);
			this.groupBox3.ResumeLayout(false);
			this.groupBox2.ResumeLayout(false);
			this.ResumeLayout(false);			
		}
		#endregion

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		static void Main() 
		{
			OutProc outProc = new OutProc();

			try{
				outProc.CreateOpcClient();
				OpcClient opcClient = outProc.OpcClient;

				OpcForm opcForm = new OpcForm(outProc);

				Application.Run(opcForm);

				opcClient = null;
			}
			catch(Exception exc)
			{
				MessageBox.Show(exc.ToString());
			}	//	end try...catch

		}	//	end Main

		private void button1_Click(object sender, System.EventArgs e) 
		{
			
			button1.Enabled = false;

			try
			{			
				textBox1.DataBindings.Add(new Binding("Text",dataControl1.GetItem("Subscription1","increment.UI1"),"ValueToString"));
				textBox2.DataBindings.Add(new Binding("Text",dataControl1.GetItem("Subscription1","increment.DATE"),"ValueToString"));
				textBox3.DataBindings.Add(new Binding("Text",dataControl1.GetItem("Subscription1","increment.BSTR array"),"ValueToString"));
				textBox4.DataBindings.Add(new Binding("Text",dataControl1.GetItem("Subscription2","maths.sin"),"ValueToString"));
				textBox5.DataBindings.Add(new Binding("Text",dataControl1.GetItem("Subscription2","time.local.second"),"ValueToString"));
			}
			catch (Exception exc)
			{
				MessageBox.Show(exc.ToString());
			}           
		}

		private void OpcForm_Closing(object sender, System.ComponentModel.CancelEventArgs e)
		{
			if (dataControl1 != null)
			{
				dataControl1.Stop();
			}   //  end if
		}  

	}	//	end class
}	//	end namespace
